hack/trivy_scan.py

```diff
@@ -0,0 +1,318 @@
+# - Summary of security counts with images a JSON file inside ../image_lists/summary_of_severity_counts_for_WG folder
+# 4. Generate a summary of the security scan reports
+# - The summary will be saved in JSON format inside ../image_lists/summary_of_severity_counts_for_WG folder
+# 5. Before run this file you have to 
+#    1. Install kustomize 
+#       - sudo apt install snapd
+#       - sudo snap install kustomize
+#    2. Install trivy
+#       - sudo apt install snapd
+#       - sudo snap install trivy
+#    4. Install Python
+#    5. Install prettytable
+#       - pip install prettytable
+
+# The script must be executed from the hack folder as it use relative paths
+
+
+import os
+import subprocess
+import re
+import argparse
+import json
+import glob
+from prettytable import PrettyTable
+
+# Dictionary mapping Kubeflow workgroups to directories containing kustomization files
+wg_dirs = {
+    "automl": "../apps/katib/upstream/installs",
+    "pipelines": "../apps/pipeline/upstream/env ../apps/kfp-tekton/upstream/env",
+    "training": "../apps/training-operator/upstream/overlays",
+    "manifests": "../common/cert-manager/cert-manager/base ../common/cert-manager/kubeflow-issuer/base ../common/istio-1-22/istio-crds/base ../common/istio-1-22/istio-namespace/base ../common/istio-1-22/istio-install/overlays/oauth2-proxy ../common/oidc-client/oauth2-proxy/overlays/m2m-self-signed ../common/dex/overlays/oauth2-proxy ../common/knative/knative-serving/overlays/gateways ../common/knative/knative-eventing/base ../common/istio-1-22/cluster-local-gateway/base ../common/kubeflow-namespace/base ../common/kubeflow-roles/base ../common/istio-1-22/kubeflow-istio-resources/base",
+    "workbenches": "../apps/pvcviewer-controller/upstream/base ../apps/admission-webhook/upstream/overlays ../apps/centraldashboard/upstream/overlays/oauth2-proxy ../apps/jupyter/jupyter-web-app/upstream/overlays ../apps/volumes-web-app/upstream/overlays ../apps/tensorboard/tensorboards-web-app/upstream/overlays ../apps/profiles/upstream/overlays ../apps/jupyter/notebook-controller/upstream/overlays ../apps/tensorboard/tensorboard-controller/upstream/overlays",
+    "serving": "../contrib/kserve - ../contrib/kserve/models-web-app/overlays/kubeflow",
+    "model-registry": "../apps/model-registry/upstream"
+}
+
+DIRECTORY = "../docs/image_lists"
+SCAN_REPORTS_DIR = os.path.join(DIRECTORY, "security_scan_reports")
+ALL_SEVERITY_COUNTS = os.path.join(DIRECTORY, "severity_counts_with_images_for_WG")
+SUMMARY_OF_SEVERITY_COUNTS = os.path.join(DIRECTORY, "summary_of_severity_counts_for_WG")
+
+os.makedirs(SCAN_REPORTS_DIR, exist_ok=True)
+os.makedirs(ALL_SEVERITY_COUNTS, exist_ok=True)
+os.makedirs(SUMMARY_OF_SEVERITY_COUNTS, exist_ok=True)
+
+def save_images(wg, images, version):
+    # Saves a list of container images to a text file named after the workgroup and version.
+    output_file = f"../docs/image_lists/kf_{version}_{wg}_images.txt"
```

Comment:
lets just create image-lists in the root and not rely on /docs, since it will be gone soon.